design/one-pager-stacks-versioning.md

```diff
@@ -0,0 +1,248 @@
+* MAJOR.MINOR.PATCH tag must not already be a released stable version else promoting this version makes no sense and
+ would conflict.
+* By convention we can not modify an existing semantic version tag
+
+### Modes for install (Implicit)
+
+Pinned Install: The user will include a fully qualified package name and a specific tag to install. In this mode we will
+ assume an external versioning system is in use and not attempt to check for updates.
+
+Channel Install: If there is a channel set and there is no tag on the package, we will install the latest from the
+ channel. The channel will be checked on an interval and the stacks controller will update the underlying stack based
+ on it's update policy and rules.
+
+### Hosted artifacts
+
+It is assumed that we will have two hosted artifacts at a channel url, i.e.
+
+Channel url:
+https://s3.amazon.com/mybucket/mychannel
+
+Which would contain:
+https://s3.amazon.com/mybucket/mychannel/latest # Latest info
+https://s3.amazon.com/mybucket/mychannel/all # All version info
+
+The stack installer would initially only use https://s3.amazon.com/mybucket/mychannel/latest, but in the future if
+ the user set an upgrade policy it would likely read from all if necessary depending on rules.
+
+### Installing from stable
+
+The stack installer will be version aware, namely it will understand either how to upgrade to the tip of a release
+ channel, or how to follow a pre-defined upgrade policy by reading the entire list of available versions.
+
+Consider the following scenario where the installer wishes to auto-upgrade to the lastest version. If the current
+ version is 1.2.3 this would be reflected in the stack install status and the package would be set to
+ `crossplane/aws-stack`.  The channel would be set to the stable release channel url hosted by s3 or other means with
+ the following valid payload from `/latest` from said channel.
+
+```json
+{
+ "name": "stable",
+ "type": "channel",
+ "package": "crossplane/aws-stack",
+ "latest":
+  {
+    "version": "1.2.3",
+    "id": "461324714c7d",
+    "createTime": "2019-09-12T17:39:04Z"
+  }
+}
+```
+
+To enact an upgrade in this scenario, you would update the payload to be a higher semantic version, i.e
+ `{"latest": {"version": 1.2.4", ...} ... }`. The installer would then install a new stack image from
+ `crossplane/aws-stack:shaofversion` by looking up the actual underlying artifact for that version in the channel.
+ The stack installer status should reflect that is installing 1.2.4, with the particular sha.
+
+If you update your release channel to something of a lesser value the stack installer would return an error that release
+ channel is of a lesser version and not upgrade. Further, if we pushed a version that contained pre-release info, this
+ would not match as a stable version and would also fail to update. Ideally we would disallow pushing invalid versions
```

Comment:
> and would also fail to update

Semi nitpick: What would be the recommended graceful degradation or workaround in this case? It would be nice if someone could install something for the first time even if `stable` appeared to be misconfigured.